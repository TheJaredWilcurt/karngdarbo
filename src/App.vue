<template>
  <div id="app" class="app-container">
    <side-bar />
    <main>
      <time-line />
      <div class="commit-diff-container">
        <commit-log />
        <file-diff />
      </div>
    </main>
  </div>
</template>

<script>
import SideBar from '@/views/SideBar.vue';
import TimeLine from '@/views/TimeLine.vue';
import CommitLog from '@/views/CommitLog.vue';
import FileDiff from '@/views/FileDiff.vue';

export default {
  name: 'App',
  components: {
    'side-bar': SideBar,
    'time-line': TimeLine,
    'commit-log': CommitLog,
    'file-diff': FileDiff
  }
};
</script>
