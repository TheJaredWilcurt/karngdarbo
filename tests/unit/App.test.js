import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex, { Store } from 'vuex';

import App from '@/App.vue';

describe('App.vue', () => {
  let store;
  let localVue;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);

    store = new Store({
      state: {
        appError: ''
      },
      getters: {},
      mutations: {},
      actions: {}
    });
  });

  describe('Created', () => {
    test('Default snapshot', () => {
      const wrapper = shallowMount(App, {
        store,
        localVue
      });

      expect(wrapper.html())
        .toMatchSnapshot();
    });
  });
});
