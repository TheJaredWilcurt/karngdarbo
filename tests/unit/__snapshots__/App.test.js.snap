// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue Created Default snapshot 1`] = `
<div id="app" class="app-container">
  <side-bar-stub></side-bar-stub>
  <main>
    <time-line-stub></time-line-stub>
    <div class="commit-diff-container">
      <commit-log-stub></commit-log-stub>
      <file-diff-stub></file-diff-stub>
    </div>
  </main>
</div>
`;
